src/routes.js

```diff
@@ -7,11 +7,25 @@ import SignupContainer from './containers/Signup';
 import HomeContainer from './containers/Home';
 import GoogleMaps from './components/common/Map.js';
 
-export default (
-  <Route path="/" component={App}>
-    <IndexRoute component={LoginContainer} />
-    <Route path="/signup" component={SignupContainer} />
-    <Route path="/home" component={HomeContainer} />
-    <Route path="/login" component={LoginContainer} />
-  </Route>
-);
+export const getRoutes = (store) => {
+  const authRequired = (nextState, replace) => {
+    const state = store.getState();
```

Comment:
Are you saving the `user` in the local storage? Because it's more lighter to access to the local storage only when the `onEnter` function is called instead of call this function every time that the `store` changes.